master/internal/provisioner/aws.go

```diff
@@ -51,21 +52,29 @@ type awsCluster struct {
 	masterURL   url.URL
 	ec2UserData []byte
 	client      *ec2.EC2
+
+	// State that is only used if spot instances are enabled
+	spot *spotState
 }
 
 func newAWSCluster(config *Config, cert *tls.Certificate) (*awsCluster, error) {
 	if err := config.AWS.initDefaultValues(); err != nil {
 		return nil, errors.Wrap(err, "failed to initialize auto configuration")
 	}
+
 	// This following AWS session is created using AWS Credentials without explicitly configuration
 	// in the code. However you need to do the following settings.
 	// See https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html
 	// 1. Use IAM roles for Amazon EC2
-	//    The following roles on any resources:
+	//    The following permissions on any resources:
 	//    "ec2:DescribeInstances",
 	//    "ec2:TerminateInstances",
 	//    "ec2:CreateTags",
 	//    "ec2:RunInstances".
+	//    If using spot instances, the following permissions will be required
```

Comment:
So I saw there are `ec2:DescribeSpotPriceHistory` in the `det-deploy` yaml files. why we don't have that here?